### PR TITLE
[KYUUBI #5408] MetadataManager tries MySQL 8 driver class first

### LINKED ISSUE #5408: [TASK][TRIVIAL] MetadataManager tries MySQL 8 driver class first
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

https://github.com/apache/kyuubi/blob/fd69c6ee1dd40b7c4f496c74a45f574220448bc6/kyuubi-server/src/main/scala/org/apache/kyuubi/server/metadata/jdbc/JDBCMetadataStore.scala#L52

If we use `com.mysql.jdbc.Driver` in the MySQL 8 JDBC driver, we will get an error output. 
If `com.mysql.cj.jdbc.Driver` exists in the classpath, we can use it first.

https://github.com/mysql/mysql-connector-j/blob/7d6b0800528b6b25c68b52dc10d6c1c8429c100c/src/legacy/java/com/mysql/jdbc/Driver.java#L42-L45

### How should we improve?

_No response_

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [X] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by lsm1**
### _Why are the changes needed?_

close #5408 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

NO

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
@lsm1 the PR itself LGTM, would you mind fixing the conflicts?

**Comment by lsm1** [comment]
> @lsm1* * the PR itself LGTM, would you mind fixing the conflicts?

ok，done

**Review by cxzl25 — APPROVED** [review]

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/metadata/jdbc/JDBCMetadataStore.scala [hunk 1]
```diff
@@ -46,10 +46,17 @@ class JDBCMetadataStore(conf: KyuubiConf) extends MetadataStore with Logging {
 
   private val dbType = DatabaseType.withName(conf.get(METADATA_STORE_JDBC_DATABASE_TYPE))
   private val driverClassOpt = conf.get(METADATA_STORE_JDBC_DRIVER)
+  private lazy val mysqlDriverClass =
+    try {
+      Class.forName("com.mysql.cj.jdbc.Driver", false, Thread.currentThread.getContextClassLoader)
+      "com.mysql.cj.jdbc.Driver"
+    } catch {
+      case _: ClassNotFoundException => "com.mysql.jdbc.Driver"
+    }
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/metadata/jdbc/JDBCMetadataStore.scala:55]: I just realized there is a method `ReflectUtils#isClassLoadable`, that can simplify the code.

**Comment by cxzl25** [comment]
Thanks, merged to master/1.8

**Pull request closed without merging**